src/test/java/com/sourcery/gymapp/backend/workout/validation/ValidationAspectTest.java

```diff
@@ -0,0 +1,106 @@
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class ValidationAspectTest {
+
+    @Mock
+    private ProceedingJoinPoint joinPoint;
+
+    @InjectMocks
+    private ValidationAspect validationAspect;
+
+    @Test
+    void shouldProceedWhenValidOrdersAreProvided() throws Throwable {
+        CreateWorkoutExerciseSetDto set1 = WorkoutExerciseSetFactory.createCreateWorkoutExerciseSetDto(1);
+        CreateWorkoutExerciseSetDto set2 = WorkoutExerciseSetFactory.createCreateWorkoutExerciseSetDto(2);
+        CreateWorkoutExerciseDto exercise1 = WorkoutExerciseFactory.createCreateWorkoutExerciseDto(1, List.of(set1, set2));
+        CreateWorkoutExerciseDto exercise2 = WorkoutExerciseFactory.createCreateWorkoutExerciseDto(2, List.of(set1, set2));
+        CreateWorkoutDto workoutDto = WorkoutFactory.createCreateWorkoutDto(null, null, List.of(exercise1, exercise2));
+
+        when(joinPoint.getArgs()).thenReturn(new Object[]{workoutDto});
+        when(joinPoint.proceed()).thenReturn("Success");
+
+        Object result = validationAspect.validateOrder(joinPoint);
+
+        assertEquals("Success", result);
+    }
+
+    @Test
+    void shouldThrowExceptionWhenOrderDoesNotStartFromOne() {
+        CreateWorkoutExerciseSetDto set1 = WorkoutExerciseSetFactory.createCreateWorkoutExerciseSetDto(2);
+        CreateWorkoutExerciseDto exercise = WorkoutExerciseFactory.createCreateWorkoutExerciseDto(1, List.of(set1));
+        CreateWorkoutDto workoutDto = WorkoutFactory.createCreateWorkoutDto(null, null, List.of(exercise));
+
+        when(joinPoint.getArgs()).thenReturn(new Object[]{workoutDto});
+
+        assertThrows(WrongOrderException.class, () -> validationAspect.validateOrder(joinPoint));
+    }
+
+    @Test
+    void shouldThrowExceptionWhenOrderIsNotSequential() {
+        CreateWorkoutExerciseSetDto set1 = WorkoutExerciseSetFactory.createCreateWorkoutExerciseSetDto(1);
+        CreateWorkoutExerciseSetDto set3 = WorkoutExerciseSetFactory.createCreateWorkoutExerciseSetDto(3);
+        CreateWorkoutExerciseDto exercise = WorkoutExerciseFactory.createCreateWorkoutExerciseDto(1, List.of(set1, set3));
+        CreateWorkoutDto workoutDto = WorkoutFactory.createCreateWorkoutDto(null, null, List.of(exercise));
+
+        when(joinPoint.getArgs()).thenReturn(new Object[]{workoutDto});
+
+        assertThrows(WrongOrderException.class, () -> validationAspect.validateOrder(joinPoint));
+    }
+
+    @Test
+    void shouldNotValidateWhenNoExercisesArePresent() throws Throwable {
```

Comment:
why is it should not?

Reply:
I meant that validation should be omitted if there are no exercises and proceed successfully with a method call.